### ISSUE TITLE
CDATA isn't parsed correctly

### ISSUE THREAD

**Issue by pqnelson** [skeeto/elfeed#192]
Hello,

I'm not certain whether this is a bug with elfeed, or with the news source in question, but if we examine [statehousenews.com](http://www.statehousenews.com/rss/rss_stories.aspx), a generic feed entry looks like:

```xml
  <item>
<title>
<![CDATA[STATE CAPITOL BRIEFS - MORNING EDITION - MONDAY, JAN. 16, 2017]]>
</title>
<link>
<![CDATA[http://www.statehousenews.com/shns/raw_asset.html?className=stories&itemName=story&selector=2017104]]>
</link>
<description>
<![CDATA[But it is going to be very interesting to see where the federal movement on this is because a lot of states now have gone towards marijuana. I would think that Jeff Sessions is against it. But at the same time you have to look at the world of people and if you are a Federalist, if you're a believer in states' rights, they may have to work in a direction that goes towards what]]>
</description>
<dc:creator>
<![CDATA[State House News Service Staff]]>
</dc:creator>
<pubDate>
<![CDATA[Mon, 16 Jan 2017 15:25:20 GMT]]>
</pubDate>
<guid>
<![CDATA[http://www.statehousenews.com/shns/raw_asset.html?className=stories&itemName=story&selector=2017104]]>
</guid>
</item>
```

Now, granted, this is terrible. Really, really terrible. But elfeed does not seem to adequately capture the link or the title. So a generic entry might look like:

```
Title: 
Date: Mon,  2 Jan 2017 08:54:27 PST
Feed: statehousenews.com
Tags: ma
Link: 

Supporters of an underground rail tunnel linking North Station and South Station in
downtown Boston have long called for Gov. Charlie Baker to come on board and claim the
rail link could at least soften the need for or scope of the long-planned expansion of South
Station. And while Baker has so far not come around to the points of view of rail link
backers like Govs. Michael Dukakis and Wil... 
```

For reasons beyond my comprehension, there is only one entry shown for `statehousenews.com` in elfeed, without a title or a link, and it never updates.

Shouldn't elfeed handle this situation? Or is something else going on "under the hood" (e.g., the RSS feed is trying to ward of bots or something)?

**Comment by skeeto** [comment]
The problem isn't so much the CDATA, since the XML parser handles it
silently. It's that having CDATA surrounded by whitespace (or anything
else) causes xml-parse-region to break the content across multiple Lisp
strings. Elfeed was latching onto the first string (just the newline
preceding the CDATA) as the link content, losing the actual link. This
should be fixed by 3be3ff0.

**Comment by pqnelson** [comment]
Hello,

Thanks for your prompt reply and patch, I pulled it down, and everything's working beautifully.

Thanks again!
